src/lua/funcs/lua_registry.rs

```diff
@@ -0,0 +1,131 @@
+//! Metamethods for accesssing the registry values!
+
+use super::super::super::registry;
+use registry::RegistryAccess;
+
+use hlua::{LuaTable};
+use hlua::any::AnyLuaValue;
+
+use rustc_serialize::json::{Json, ToJson};
+use std::ops::Deref;
+pub fn index(table: AnyLuaValue, lua_key: AnyLuaValue)
+             -> Result<AnyLuaValue, &'static str> {
+    if let AnyLuaValue::LuaString(key) = lua_key {
+        let maybe_json = registry::get_json(&key);
+        if let Some(json_pair) = maybe_json {
+            let (access, json) = json_pair;
+            if access == RegistryAccess::Private {
+                // TODO In lua_async: return nil
+                Err("No value found for that key!")
+            }
+            else {
+                Ok(convert_json(json.deref().to_json()))
+            }
+        }
+        else {
+            Err("No value found for that key!")
+        }
+    }
+    else {
+        Err("The registry can only be indexed with keys!")
+    }
+}
+
+// Prevent lua from changing the registry?
+pub fn new_index(table: AnyLuaValue, key: AnyLuaValue, val: AnyLuaValue)
```

Comment:
If we don't have an implementation for that yet, perhaps we should just leave it out for now so we don't need to remove it later? We can always add it in later.